src/lib/containers/useImmutableTableQuery.ts

```diff
@@ -0,0 +1,171 @@
+import { useCallback, useEffect, useState } from 'react'
+import { QueryBundleRequest } from '../utils/synapseTypes/Table/QueryBundleRequest'
+import { cloneDeep } from 'lodash-es'
+import * as DeepLinkingUtils from '../utils/functions/deepLinkingUtils'
+import { DEFAULT_PAGE_SIZE } from '../utils/SynapseConstants'
+import { parseEntityIdAndVersionFromSqlStatement } from '../utils/functions/sqlFunctions'
+
+export type ImmutableTableQueryResult = {
+  /** The ID of the table parsed from the SQL query */
+  entityId: string
+  /** The version number of the table parsed from the SQL query */
+  versionNumber?: number
+  getInitQueryRequest: () => QueryBundleRequest
+  getLastQueryRequest: () => QueryBundleRequest
+  setQuery: (
+    queryRequest:
+      | QueryBundleRequest
+      | ((lastQueryRequest: QueryBundleRequest) => QueryBundleRequest),
+  ) => void
```

Comment:
You can now update the query with a function that transforms a mutable clone of the current request. This would be more useful if we returned an immutable version of the query request as above.

```suggestion
  setQuery: (
    queryRequest:
      | QueryBundleRequest
      | ((lastQueryRequest: QueryBundleRequest) => QueryBundleRequest),
  ) => void
```